iceoryx_posh/test/mocks/chunk_mock.hpp

```diff
@@ -32,10 +32,19 @@ class ChunkMock
     {
 #if defined(QNX) || defined(QNX__) || defined(__QNX__)
         m_rawMemory = static_cast<uint8_t*>(memalign(Alignment, Size));
+        m_alignedMemory = m_rawMemory;
+#elif defined(__APPLE__)
+        m_rawMemory = static_cast<uint8_t*>(malloc(Size + Alignment));
+        auto memAddress = reinterpret_cast<uint64_t>(m_rawMemory);
+        auto offset = memAddress % Alignment;
+        auto alignedAddress = offset = 0 ? memAddress : memAddress + Alignment - offset;
+        m_alignedMemory = reinterpret_cast<uint8_t*>(alignedAddress);
 #elif defined(_WIN32)
         m_rawMemory = static_cast<uint8_t*>(_aligned_malloc(Alignment, Size));
+        m_alignedMemory = m_rawMemory;
 #else
         m_rawMemory = static_cast<uint8_t*>(aligned_alloc(Alignment, Size));
+        m_alignedMemory = m_rawMemory;
 #endif
         assert(m_rawMemory != nullptr && "Could not get aligned memory");
 
```

Comment:
memset and placement new for m_chunkHeader must use the m_alignedMemory member. I think it would also be sufficient if m_alignedMemory would be a local variable.

Reply:
fixed by removing all platform macros and introduced `alignedAlloc` and a corresponding `alignedFree` in cxx helpless